Sources/RemoteNotification.swift

```diff
@@ -33,28 +33,39 @@ public struct RemoteNotification: CustomStringConvertible, Equatable {
     /// The identifier for the UIUserNotificationCategory associated with the push notification. Derived from userInfo["category"]
     public let categoryIdentifier: String?
     /// Custom fields on the remote notification.
-    public let userInfo: [String : AnyObject]
+    public let userInfo: [String : Any]
     /// The dictionary representation of the remote notification.
-    public let remoteNotificationDictionary: [String : AnyObject]
+    public let remoteNotificationDictionary: [String : Any]
     
-    public init?(remoteNotification: [NSObject : AnyObject]?) {
-        guard let remoteNotification = remoteNotification as? [String : AnyObject] else {
+    // MARK: Equatable
+    
+    public static func ==(lhs: RemoteNotification, rhs: RemoteNotification) -> Bool {
+        return lhs.alert == rhs.alert
+            && lhs.badge == rhs.badge
+            && lhs.sound == rhs.sound
+            && lhs.contentAvailable == rhs.contentAvailable
+            && lhs.categoryIdentifier == rhs.categoryIdentifier
+            && NSDictionary(dictionary: lhs.userInfo) == NSDictionary(dictionary: rhs.userInfo)
+    }
+    
+    public init?(remoteNotification: [AnyHashable : Any]?) {
+        guard let remoteNotification = remoteNotification as! [String : Any]? else {
             return nil
         }
         
         // Parse the notification payload.
         // See https://developer.apple.com/library/ios/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/Chapters/TheNotificationPayload.html for more information on expected values.
         alert = Alert(remoteNotification: remoteNotification)
         
-        let apnsDictionary = remoteNotification[APSServiceKey] as? [String : AnyObject]
-        badge = apnsDictionary?[badgeKey] as? Int
-        sound = apnsDictionary?[soundKey] as? String
+        let apnsDictionary = remoteNotification[apsServiceKey] as! [String : Any]?
```

Comment:
🤔. There's a balancing act to be done here.

Pros of crashing:
- Developers developing against this API are going to want to know, fast, when they're doing it wrong.
- We're going to want to know on beta iOS builds that our assumptions are wrong

Cons of crashing:
- A new iOS build could catch a dev by surprise and we could start crashing.

I think the risk of crashing here is pretty low – these keys have been stable since push notifications were introduced, which pushes me towards thinking that `!` is the right approach here.

Thoughts @mikelikespie, @efirestone, @martinmroz? Would love some more input here.